XenAdmin/Controls/MultipleDvdIsoList.cs

```diff
@@ -251,17 +251,26 @@ private void comboBoxDrive_SelectedIndexChanged(object sender, EventArgs e)
             cdChanger1.Drive = (comboBoxDrive.SelectedItem as VbdCombiItem)?.Vbd;
         }
 
-
         private void newCDLabel_Click(object sender, EventArgs e)
         {
-            if (VM != null)
-            {
-                var createDriveAction = new CreateCdDriveAction(VM);
-                createDriveAction.ShowUserInstruction += CreateDriveAction_ShowUserInstruction;
+            if (VM == null)
+                return;
 
-                using (var dlg = new ActionProgressDialog(createDriveAction, ProgressBarStyle.Marquee))
-                    dlg.ShowDialog(this);
+            if (VM.IsHVM())
+            {
+                using (var dialog = new WarningDialog(
+                           string.Format(Messages.NEW_DVD_DRIVE_CREATE_CONFIRMATION, VM.Name()),
+                           new ThreeButtonDialog.TBDButton(Messages.NEW_DVD_DRIVE_CREATE_YES_BUTTON, DialogResult.Yes, ThreeButtonDialog.ButtonType.ACCEPT, true),
+                           ThreeButtonDialog.ButtonNo))
+                    if (dialog.ShowDialog(Program.MainWindow) != DialogResult.Yes)

```

Comment:
I understood that we wanted the confirmation dialog in all cases here. I thought we were adding it to avoid users inadvertently creating things in general. Not sure if the hot-plugging issue applies to VMs without `HVM_boot_policy` but we can just show a generic confirmation dialog anyway there. thoughts?

Reply:
I added the check because in CreateCdDriveAction the VBD is created with `empty=true` and later VbdCreateAndPlugAction does nothing if VBD.empty and the VM is not HVM - to my understanding if it's not HVM it won't try to hot plug it, so no need to say it can't be hot unplugged. Note also that in recent servers all VMs are HVMs, so it is less likely that the user will have a non-HVM.